argo/providers/base/provider_app.go

```diff
@@ -9,17 +9,25 @@ import (
 )
 
 var ProviderConstants = struct {
-	PipelineDefinitionParameter string
-	ProviderConfigParameter     string
-	PipelineIdParameter         string
-	PipelineFileParameter       string
-	OutputParameter             string
+	PipelineDefinitionParameter         string
+	ExperimentDefinitionParameter       string
+	RunConfigurationDefinitionParameter string
+	ProviderConfigParameter             string
+	PipelineIdParameter                 string
+	ExperimentIdParameter               string
+	RunConfigurationIdParameter         string
+	PipelineFileParameter               string
+	OutputParameter                     string
 }{
-	PipelineDefinitionParameter: "pipeline-definition",
-	ProviderConfigParameter:     "provider-config",
-	PipelineIdParameter:         "pipeline-id",
-	PipelineFileParameter:       "pipeline-file",
-	OutputParameter:             "out",
+	PipelineDefinitionParameter:         "pipeline-definition",
+	ExperimentDefinitionParameter:       "experiment-definition",
+	RunConfigurationDefinitionParameter: "runconfiguration-definition",
+	ProviderConfigParameter:             "provider-config",
+	PipelineIdParameter:                 "pipeline-id",
+	ExperimentIdParameter:               "experiment-id",
+	RunConfigurationIdParameter:         "runconfiguration-id",
+	PipelineFileParameter:               "pipeline-file",
```

Comment:
Guess these will be renamed to `resource-definition`, `resource-id`, `resource-file` later in this PR after refactoring. We should be able to create generic workflow steps for create/update with and without file and a single delete.

Reply:
Raised #178.